features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/dialogs/PurchaseGiftCardConfirmDialog.kt

```diff
@@ -0,0 +1,126 @@
+import org.dash.wallet.features.exploredash.data.model.GiftCardDetailsDialogModel
+import org.dash.wallet.features.exploredash.databinding.DialogConfirmPurchaseGiftCardBinding
+import org.dash.wallet.features.exploredash.ui.ExploreViewModel
+import java.util.*
+
+@FlowPreview
+@ExperimentalCoroutinesApi
+@AndroidEntryPoint
+class PurchaseGiftCardConfirmDialog : OffsetDialogFragment() {
+    @StyleRes
+    override val backgroundStyle = R.style.PrimaryBackground
+
+    private val exploreViewModel: ExploreViewModel by navGraphViewModels(R.id.explore_dash) { defaultViewModelProviderFactory }
+
+    private val binding by viewBinding(DialogConfirmPurchaseGiftCardBinding::bind)
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.dialog_confirm_purchase_gift_card, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val merchent = exploreViewModel.purchaseGiftCardData?.second
+        val paymentValue = exploreViewModel.purchaseGiftCardData?.first
+        merchent?.let {
+            binding.merchentName.text = it.name
+            merchent.logoLocation?.let {
+                Glide.with(requireContext())
+                    .load(it)
+                    .placeholder(org.dash.wallet.common.R.drawable.ic_image_placeholder)
+                    .error(org.dash.wallet.common.R.drawable.ic_image_placeholder)
+                    .transition(DrawableTransitionOptions.withCrossFade(200))
+                    .into(binding.merchentLogo)
+            }
+            binding.giftCardDiscountValue.text = "0%"
+        }
+
+        paymentValue?.let {
+            binding.giftCardTotalValue.text =
+                GenericUtils.fiatToString(it.second)
+
+            binding.giftCardYouPayValue.text =
+                GenericUtils.fiatToString(it.second)
+
+            binding.purchaseCardValue.text =
+                GenericUtils.fiatToString(it.second)
+        }
+
+        binding.collapseButton.setOnClickListener {
+            dismiss()
+        }
+
+        binding.confirmButton.setOnClickListener {
+            lifecycleScope.launch {
+                val merchant = exploreViewModel.purchaseGiftCardData?.second
+                val paymentValue = exploreViewModel.purchaseGiftCardData?.first
+                merchant?.merchantId?.let {
+                    paymentValue?.let { amountValue ->
+                        exploreViewModel.purchaseGiftCard(
+                            merchantId = it,
+                            giftCardAmount = amountValue.first.toPlainString().toDouble(),
+                            currency = Constants.DASH_CURRENCY,
+                            deviceID = UUID.randomUUID().toString()
+                        )
+                    }
+                }
```

Comment:
This code doesn't need to be in the fragment. You're getting two values from the viewModel and calling a method in the same viewModel.
Just move this whole thing to the viewModel.